src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -1417,6 +1418,28 @@ static QVariant fcnAttribute( const QVariantList &values, const QgsExpressionCon
   return feature.attribute( attr );
 }
 
+static QVariant fcnAttributes( const QVariantList &values, const QgsExpressionContext *context, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QgsFeature feature;
+  QString attr;
+  if ( values.size() == 0 || values.at( 0 ).isNull() )
+  {
+    feature = context->feature();
+  }
+  else
+  {
+    feature = QgsExpressionUtils::getFeature( values.at( 0 ), parent );
+  }
+
+  const QgsFields fields = !context->fields().isEmpty() ? context->fields() : feature.fields();
```

Comment:
Hmm, I think that's incorrect. If you pass on a feature through parameter, you'd _always_ want that feature's fields to be used over the context, no?

Reply:
Ah - you're right!